.github/workflows/main-auto-par.yml

```diff
@@ -75,3 +75,8 @@ jobs:
       - name: Autotools Build
         run: make -j3
         working-directory: ${{ runner.workspace }}/build
+
+      - name: Autotools Run Tests
+        run: make check -j2
```

Comment:
For these actions we don't want to run the tests in parallel since that may cause slowdowns when the actual parallel tests run. Really what we want to do is run the serial tests in parallel separately and then run the parallel tests serially and set `RUNPARALLEL` for autotools and `MPIEXEC_MAX_NUMPROCS=X` (probably X=2) for CMake.

Reply:
I've thought about that, but that's more complicated since the serial tests are in multiple directories

Reply:
> I've thought about that, but that's more complicated since the serial tests are in multiple directories

I'd imagine you can just do `cd test; make check -j2; cd ../tools/test; make check -j2` to only run all the serial tests that are interesting. It's not future-proof, but we can always start with just CMake like mentioned.

Reply:
I'll update this when I get in on Tuesday